### ISSUE TITLE
no function clause matching in NewRelic.Tracer.Macro.__on_definition__/6

### ISSUE THREAD

**Issue by mirjoy** [newrelic/elixir_agent#254]
**Describe the bug**
When adding NewRelic.Tracer to several modules, I have gotten the error `no function clause matching in NewRelic.Tracer.Macro.__on_definition__/6`

The unmatched argument is the sixth argument, as the keyword list includes not only `:do` and `:rescue` but also `:after` - I've added an example below.

I'm able to handle the error by adding this function clause to `NewRelic.Tracer.Macro`
```
  def __on_definition__(env, _access, name, _args, _guards, do: _, rescue: _, after: _) do
    Logger.warn(
      "[New Relic] Unable to trace `#{inspect(env.module)}.#{name}` due to top-level rescue clause -- please remove @trace"
    )
  end
```
Although obviously this is a brittle solution. Some of my other methods are coming in with Keyword lists that include [:do, :catch], etc. 

Unmatched argument

```
[
    do:
      {:with, [line: 55],
       [
         {:<-, [line: 55],
          [
            {:ok, {:contents, [line: 55], nil}},
            {{:., [line: 55], [{:__aliases__, [line: 55], [:File]}, :read]}, [line: 55],
             [{:path, [line: 55], nil}]}
          ]},
         {:<-, [line: 56],
          [
            [{:|, [line: 56], [{:headers, [line: 56], nil}, {:body, [line: 56], nil}]}],
            {{:., [line: 56], [{:__aliases__, [line: 56], [:CSVParser]}, :parse]}, [line: 56],
             [{:contents, [line: 56], nil}]}
          ]},
         {:<-, [line: 57],
          [:ok, {:check_headers_are_valid_binaries, [line: 57], [{:headers, [line: 57], nil}]}]},
         {:<-, [line: 58],
          [
            :ok,
            {:check_required_fields, [line: 58],
             [{:headers, [line: 58], nil}, {:company, [line: 58], nil}]}
          ]},
         [
           do:
             {:__block__, [],
              [
                {:=, [line: 60],
                 [
                   {:user_list, [line: 60], nil},
                   {:|>, [line: 65],
                    [
                      {:|>, [line: 64],
                       [
                         {:|>, [line: 63],
                          [
                            {:|>, [line: 62],
                             [
                               {:body, [line: 61], nil},
                               {:map_positions_with_row_data, [line: 62],
                                [{:headers, [line: 62], nil}]}
                             ]},
                            {{:., [line: 63],
                              [
                                {:__aliases__, [line: 63], [:SharedFormatter]},
                                :fix_users_key_formatting
                              ]}, [line: 63], [{:company, [line: 63], nil}]}
                          ]},
                         {{:., [line: 64], [{:__aliases__, [line: 64], [:Enum]}, :map]},
                          [line: 64],
                          [
                            {:&, [line: 64],
                             [{:/, [line: 64], [{:scrub_emailaddress, [line: 64], nil}, 1]}]}
                          ]}
                       ]},
                      {{:., [line: 65], [{:__aliases__, [line: 65], [:Enum]}, :map]}, [line: 65],
                       [
                         {:&, [line: 65],
                          [{:/, [line: 65], [{:scrub_archive, [line: 65], nil}, 1]}]}
                       ]}
                    ]}
                 ]},
                {:ok, {:user_list, [line: 67], nil}}
              ]},
           else: [
             {:->, [line: 69], [[[]], {:@, [line: 69], [{:invalid_csv_error, [line: 69], nil}]}]},
             {:->, [line: 70],
              [
                [
                  {:=, [line: 70],
                   [{:error, {:_msg, [line: 70], nil}}, {:error, [line: 70], nil}]}
                ],
                {:error, [line: 70], nil}
              ]}
           ]
         ]
       ]},
    rescue: [
      {:->, [line: 73],
       [
         [
           {:in, [line: 73],
            [{:err, [line: 73], nil}, {:__aliases__, [line: 73], [:NimbleCSV, :ParseError]}]}
         ],
         {:error, {{:., [line: 73], [{:err, [line: 73], nil}, :message]}, [line: 73], []}}
       ]},
      {:->, [line: 74],
       [
         [{:__aliases__, [line: 74], [:FunctionClauseError]}],
         {:@, [line: 74], [{:invalid_csv_error, [line: 74], nil}]}
       ]}
    ],
    after:
      {{:., [line: 76], [{:__aliases__, [line: 76], [:File]}, :rm]}, [line: 76],
       [{:path, [line: 76], nil}]}
  ]
```

**Environment**
* Elixir & Erlang version (`elixir -v`): Elixir 1.9.2 (compiled with Erlang/OTP 22)
* Agent version (`mix deps | grep new_relic_agent`): 1.19.2

**Comment by binaryseed** [comment]
Thanks, I'll get a fix in

**Comment by binaryseed** [comment]
Fix in https://github.com/newrelic/elixir_agent/issues/254

**Comment by mirjoy** [comment]
Thanks!

**Comment by binaryseed** [comment]
Released in `1.19.4`
